includes/Core/Authentication/Has_Multiple_Admins.php

```diff
@@ -65,9 +80,82 @@ public function get() {
 			$user_query   = new WP_User_Query( $user_query_args );
 			$admins_count = $user_query->get_total();
 
-			$this->transients->get( self::OPTION, $admins_count, HOUR_IN_SECONDS );
+			$this->transients->set( self::OPTION, $admins_count, WEEK_IN_SECONDS );
 		}
 
 		return $admins_count > 1;
 	}
+
+	/**
+	 * Handles user deletion.
+	 *
+	 * Executed by the `deleted_user` hook.
+	 * We skip clearing the transient cache, only if we are sure that the user
+	 * being deleted is not an admin. The $user parameter is only available
+	 * in WP 5.5+, so we do not rely on it.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param int     $user_id User ID.
+	 * @param bool    $reassign Whether the user's posts are being reassigned.
+	 * @param WP_User $user    User object.
+	 * @return void
+	 */
+	public function handle_user_deletion( $user_id, $reassign, $user = null ) {
+		if ( isset( $user ) && ! in_array( 'administrator', $user->roles, true ) ) {
+			return;
+		}
+		$this->transients->delete( self::OPTION );
+	}
+
+	/**
+	 * Clears transient cache when a user is added or updated
+	 *
+	 * Executed by the `set_user_role` hook. `WP_User::set_role()` is used when
+	 * adding a new user as well as updating an existing user's profile (so we do not
+	 * have to specifically hook into `user_register` or `profile_update` hooks).
+	 * Furthermore, `WP_User::set_role()` calls `remove_role()` and `add_role()` internally,
+	 * but only after WordPress 6.0. So we still have to hook into `set_user_role` to cover
+	 * all versions before it.
+	 * We skip clearing the transient cache only if we are sure that we aren't
+	 * changing a role from or to the 'administrator' role.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param int    $user_id   User ID.
+	 * @param string $role      New role.
+	 * @param array  $old_roles Old roles.
+	 * @return void
+	 */
+	public function handle_set_user_role( $user_id, $role, $old_roles = array() ) {
+		if ( ! in_array( 'administrator', (array) $old_roles, true ) && 'administrator' !== $role ) {
+			return;
+		}
+		$this->transients->delete( self::OPTION );
+	}
+
+	/**
+	 * Handles user role changes.
+	 *
+	 * Executed by the `add_user_role` and `remove_user_role` hooks. These hooks
+	 * are called internally by `WP_User::add_role()` and `WP_User::remove_role()`
+	 * respectively. From WordPress 6.0, these hooks are always called even
+	 * when the more generic `WP_User::set_role()` is used when adding or updating
+	 * a user. However, for backwards compatibility with earlier versions, we still
+	 * have to clear the cache and hook into `set_user_role` separately.
+	 * We skip clearing the transient cache only if we are sure that
+	 * the role being added/removed is 'administrator'.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param int    $user_id User ID.
+	 * @param string $role    Role being added/removed.
+	 * @return void
+	 */
+	public function handle_add_remove_role( $user_id, $role ) {
+		if ( isset( $role ) && 'administrator' !== $role ) {
```

Comment:
Do we need the `isset` here? I think we can simplify it to just the equality check, no?